pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConfiguration.java

```diff
@@ -146,46 +145,38 @@ public void setZookeeperSessionTimeoutMs(int zookeeperSessionTimeoutMs) {
         this.zookeeperSessionTimeoutMs = zookeeperSessionTimeoutMs;
     }
 
-    public int getServicePort() {
+    public Integer getServicePort() {
```

Comment:
I prefer to use `Optional<Integer>` here to force the caller to explicitly handle the fact that this can be null.